src/domain/challenge/ecoverse/ecoverse.service.ts

```diff
@@ -113,37 +139,33 @@ export class EcoverseService {
     return ecoverse;
   }
 
-  async getDefaultEcoverseOrFail(
-    options?: FindOneOptions<Ecoverse>
-  ): Promise<IEcoverse> {
-    const ecoverseId = await this.getEcoverseId(); // todo - remove when can have multiple ecoverses
-    return await this.getEcoverseOrFail(ecoverseId, options);
+  async getChallenges(ecoverse: IEcoverse): Promise<IChallenge[]> {
+    const ecoverseWithChallenges = await this.getEcoverseByIdOrFail(
+      ecoverse.id,
+      {
+        relations: ['challenges'],
+      }
+    );
+    return ecoverseWithChallenges.challenges || [];
   }
 
-  async getEcoverseId(): Promise<number> {
-    const ecoverse = await this.ecoverseRepository
-      .createQueryBuilder('ecoverse')
-      .select('ecoverse.id')
-      .getOne(); // TODO [ATS] Replace with getOneOrFail when it is released. https://github.com/typeorm/typeorm/blob/06903d1c914e8082620dbf16551caa302862d328/src/query-builder/SelectQueryBuilder.ts#L1112
-
-    if (!ecoverse) {
-      throw new ValidationException(
-        'Ecoverse is missing!',
-        LogContext.BOOTSTRAP
+  async getOpportunities(ecoverse: IEcoverse): Promise<IOpportunity[]> {
+    const opportunities: IOpportunity[] = [];
+    const challenges = await this.getChallenges(ecoverse);
+    for (let i = 0; i < challenges.length; i++) {
```

Comment:
for(const entity of entities)... --> like everywhere else in the codebase

Reply:
Agreed. Still default to the old style at times :)